subprojects/platform-native/src/main/java/org/gradle/nativeplatform/MachineArchitecture.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.Named;
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.tasks.Input;
+
+/**
+ * Represents the architecture of a binary.  Typical architectures include "x86", "x64", "ia64", and "arm".
+ *
+ * @since 5.1
+ */
+@Incubating
+public abstract class MachineArchitecture implements Named {
+    public static final Attribute<MachineArchitecture> ARCHITECTURE_ATTRIBUTE = Attribute.of("org.gradle.native.architecture", MachineArchitecture.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Input
+    public abstract String getName();
+
+    /**
+     * The intel x86 32-bit architecture
+     */
+    public static final String X86 = "x86";
+
+    /**
+     * The intel x86 64-bit architecture
+     */
+    public static final String X64 = "x86-64";
+
+    /**
+     * The Itanium 64-bit architecture
+     */
+    public static final String IA64 = "ia-64";
```

Comment:
Itanium is dead, SPARC is mostly dead, so that leaves PPC and ARM.  Since we don't support cross-compilation out-of-the-box for these plugins and I doubt many/any people are running Gradle on those machines, I would say we should ditch them since it's only going to be tempting to use them.

Reply:
Ok, I've pulled them out.